tests/components/automation/test_init.py

```diff
@@ -700,6 +703,13 @@ def running_cb(event):
             {ATTR_ENTITY_ID: entity_id, automation.CONF_STOP_ACTIONS: False},
             blocking=True,
         )
+    elif service == "stop":
+        await hass.services.async_call(
+            automation.DOMAIN,
+            SERVICE_STOP,
+            {ATTR_ENTITY_ID: entity_id},
+            blocking=True,
+        )
```

Comment:
Is it possible to test that the automation actually stopped?

Reply:
That's what the existing code already does :) I tried by removing my actual code (so the `await self.action_script.async_stop()`) and the test did fail as expected.

How I read the code it starts the automation, which will then block / wait in the `wait_template`, at this point the service (`turn_off` / `stop` / `reload`) is invoked which should thus stop the automation. Afterwards the state is changed to the waiting condition so the actions, if still running (`turn_off` with `stop_actions: false`), will proceed after which the `test.automation` service is invoked. The test then validates the passed in `calls` list which should be empty if the automation stopped and contain one item if it didn't stop. Which I presume will be populated by the `test.automation` call.

So for `automation.stop` it will block in the `wait_template` and because of the `automatio.stop` it will thus never leave the `wait_template`, thus `test.automation` is never invoked and the `calls` list stays empty.

Reply:
Ok, I believe you!  But to be honest I don't understand it.

I was expecting to see something like
- Start an automation that won't finish instantly
- Verify that the automation is running
- Trigger the new stop action
- Verify that the automation is stopped

The len(calls) check with the conditional quantity might do the same thing, but it's not obvious (at least to me!)

Reply:
I had to check it multiple times as well :) But it does work fine I guess. As stated, I did remove the implementation of the service and the test failed as expected.

And I do believe it works as you describe:
> Start an automation that won't finish instantly

This is what the `wait_template` does

> Verify that the automation is running

Digging into it I believe this is what the `running` event is used for. After firing the event / trigger further execution of the unit test waits for this event, thus making sure the automation started.

> Trigger the new stop action

That's obviously the code I added

> Verify that the automation is stopped

That would be the `test.automation` service invocation in combination with the `len(calls)` assertion I guess, as described above.